src/serializers/fields.rs

```diff
@@ -200,7 +200,7 @@ impl TypeSerializer for GeneralFieldsSerializer {
                     let value = infer_to_python(value, next_include, next_exclude, &extra)?;
                     output_dict.set_item(key, value)?;
                 } else if extra.check.enabled() {
-                    return Err(PydanticSerializationUnexpectedValue::new_err(None));
+                    // return Err(PydanticSerializationUnexpectedValue::new_err(None));
```

Comment:
I think this line is erroring if there is an unexpected extra field, and we shouldn't do that if we want to be able to serialize subclasses (which might have extra fields).